### ISSUE TITLE
feat: add error handling for logging out

### ISSUE THREAD

**Issue by matthewkurtti** [ourjapanlife/findadoc-web#1500]
# Problem
Currently, the log out function in authStore does not handle potential errors that could occur. This will be necessary for giving error messages to users so that they can know what went wrong if logging out doesn't work

# Success

Log out function in authStore catches errors and provides detailed error messages

### Requirements

- [ ] Log out function in authStore catches errors
- [ ] Provides detailed error messages

# Implementation notes
This one will be handled by @NabbeunNabi because it'll involve auth0
